File: articles/perseverance-of-effort-versus-consistency-of-interests/c/grit_facet_score.c
#include <stdio.h>

double grit_total_score(double perseverance_effort, double consistency_interests) {
    return 0.60 * perseverance_effort + 0.40 * consistency_interests;
}

int main(void) {
    double perseverance_effort = 0.82;
    double consistency_interests = 0.71;
    printf("Synthetic grit total score: %.3f\n",
           grit_total_score(perseverance_effort, consistency_interests));
    return 0;
}

File: articles/perseverance-of-effort-versus-consistency-of-interests/go/grit_facet_score.go
package main

import "fmt"

func GritTotalScore(perseveranceEffort float64, consistencyInterests float64) float64 {
	return 0.60*perseveranceEffort + 0.40*consistencyInterests
}

func main() {
	fmt.Printf("Synthetic grit total score: %.3f\n", GritTotalScore(0.82, 0.71))
}
